src/safeposix/syscalls/net_calls.rs

```diff
@@ -2109,51 +2169,69 @@ impl Cage {
                         }
                     }
                 } else {
-                    //Reaching this means that the socket is closed. Removing the sockobj
-                    //indicates that the sockobj will drop, and therefore close
+                    // Reaching this means that the socket is closed after close_syscall. Removing
```

Comment:
where was close_syscall called?

Reply:
this function could also be called when the sockethandle is dropped. See net.rs. Drop happens when sockethandle is not referenced by anyone, which will happen after close_syscall